samples/sample_common/include/sample_utils.h

```diff
@@ -299,6 +299,32 @@ struct IsPairedMfxExtBuffer : std::false_type {};
 template <>
 struct IsPairedMfxExtBuffer<mfxExtAVCRoundingOffset> : std::true_type {};
 
+template <typename R>
+struct ExtParamAccessor
+{
+private:
+    using mfxExtBufferDoublePtr = mfxExtBuffer**;
+public:
+    mfxU16& NumExtParam;
+    mfxExtBufferDoublePtr& ExtParam;
+    ExtParamAccessor(const R& r):
+        NumExtParam(const_cast<mfxU16&>(r.NumExtParam)),
+        ExtParam(const_cast<mfxExtBufferDoublePtr&>(r.ExtParam)) {}
+};
+
+template <>
+struct ExtParamAccessor<mfxFrameSurface1>
```

Comment:
Why do you need this specialization?

You can just use generic template for ```mfxFrameData```

```
auto accessor = ExtParamAccessor(surface.Data);
```

as for me it looks more starightforward

Reply:
@mgonchar , Please show how "auto accessor = ExtParamAccessor(surface.Data)" will look like in generic version of "ExtBufHolder& operator=(const T& ref)"

Reply:
@mgonchar, ping ..

Reply:
ok, I figured out what's happening